src/Microsoft.TemplateSearch.TemplateDiscovery/PackChecking/PackSourceChecker.cs

```diff
@@ -30,65 +30,89 @@ public PackSourceChecker(IPackProvider packProvider, PackPreFilterer packPreFilt
         public async Task<PackSourceCheckResult> CheckPackagesAsync()
         {
             List<PackCheckResult> checkResultList = new List<PackCheckResult>();
+            HashSet<IPackInfo> alreadySeenPacks = new HashSet<IPackInfo>();
             HashSet<string> alreadySeenTemplateIdentities = new HashSet<string>();
 
-            int count = 0;
-            Console.WriteLine($"{await _packProvider.GetPackageCountAsync().ConfigureAwait(false)} packs discovered, starting processing");
-
-            await foreach (IInstalledPackInfo packInfo in _packProvider.GetCandidatePacksAsync().ConfigureAwait(false))
+            foreach (IPackProvider packProvider in _packProviders)
             {
-                PackCheckResult preFilterResult = PrefilterPackInfo(packInfo);
-                if (preFilterResult.PreFilterResults.ShouldBeFiltered)
-                {
-                    Verbose.WriteLine($"{packInfo.Id}::{packInfo.Version} is skipped, {preFilterResult.PreFilterResults.Reason}");
-                    checkResultList.Add(preFilterResult);
-                }
-                else
+                Console.WriteLine($"Processing pack provider {packProvider.Name}:");
+                int count = 0;
+                Console.WriteLine($"{await packProvider.GetPackageCountAsync().ConfigureAwait(false)} packs discovered, starting processing");
+
+                await foreach (IPackInfo sourceInfo in packProvider.GetCandidatePacksAsync().ConfigureAwait(false))
```

Comment:
Woot, `await foreach` I need to brush my C# skills it seems, never seen this...
Ignore this comment :)